code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -33,23 +33,88 @@
 	var/next_dest
 	var/next_dest_loc
 
+	var/obj/item/weapon
+	var/weapon_orig_force = 0
+
+	var/list/stolen_valor
+	var/orig_name
+
+	var/list/officers = list("Admiral", "Special Ops Officer", "Captain", "Head of Personnel", "Death Commando", "Head of Security")
+	var/list/command = list("Admiral" = "Admiral", "Special Ops Officer" = "Col.", "Captain" = "Cpt.","Head of Personnel" = "Lt.")
+	var/list/security = list("Death Commando" = "Trooper", "Head of Security" = "Maj.", "Warden" = "Sgt.", "Detective" =  "Det.", "Security Officer" = "Officer")
+	var/list/engineering = list("Chief Engineer" = "Chief Engineer", "Station Engineer" = "Engineer", "Atmospherics Technician" = "Technician")
+	var/list/medical = list("Chief Medical Officer" = "C.M.O.", "Medical Doctor" = "M.D.", "Chemist" = "Pharm.D.")
+	var/list/research = list("Research Director" = "Ph.D.", "Roboticist" = "M.S.", "Scientist" = "B.S.")
+	var/list/legal = list("Lawyer" = "Esq.")
+
+	var/list/prefixes
+	var/list/suffixes
+
+
+/mob/living/simple_animal/bot/cleanbot/proc/deputize(obj/item/W, mob/user)
+	if(in_range(src, user))
+		to_chat(user, "<span class='notice'>You attach \the [W] to \the [src].</span>")
+		user.transferItemToLoc(W, src)
+		weapon = W
+		weapon_orig_force = weapon.force
+		weapon.force = weapon.force / 2
+		icon_state = "cleanbot[on]" + (weapon ? "-armed" : "")
+
+/mob/living/simple_animal/bot/cleanbot/proc/update_titles()
+	var/working_title = ""
+
+	for(var/pref in prefixes)
+		for(var/title in pref)
+			if(title in stolen_valor)
+				working_title += pref[title] + " "
+				if(title in officers)
+					commissioned = TRUE
+				break
+
+	working_title += orig_name
+
+	for(var/suf in suffixes)
+		for(var/title in suf)
+			if(title in stolen_valor)
+				working_title += " " + suf[title]
+				break
+
+	name = working_title
+
+/mob/living/simple_animal/bot/cleanbot/examine(mob/user)
+	. = ..()
+	if(weapon)
+		. += " <span class='warning'>Is that \a [weapon] taped to it...?</span>"
+
 /mob/living/simple_animal/bot/cleanbot/Initialize()
 	. = ..()
 	get_targets()
-	icon_state = "cleanbot[on]"
+	icon_state = "cleanbot[on]" + (weapon ? "-armed" : "")
 
 	var/datum/job/janitor/J = new/datum/job/janitor
 	access_card.access += J.get_access()
 	prev_access = access_card.access
+	orig_name = name
```

Comment:
Rather than storing a var you can just use `initial(name)`

Reply:
Done